spec/index.md

```diff
@@ -5536,8 +5623,76 @@ S with
 
 ```
 
+#### Trimming canister logs
+
+Canister logs can be trimmed if the total length of their contents exceeds 4KiB.
+
+Conditions
+
+```html
+
+S.canister_logs[CanisterId] = Older_logs · Newer_logs
+SUM { |l.content| | l <- Older_logs } > 4KiB
```

Comment:
The implementation uses the size of the log entry struct in Rust here instead of the binary size of the content.

Reply:
I.e., the log entry size in the replica implementation is a constant away from the size specified here.

Reply:
yes, otherwise it can have unlimited number of entries with zero content size.

Reply:
The relevant function in the replica code is [here](https://github.com/dfinity/ic/blob/8176ea68ae82cb2ba3eec346259e00bf318a171d/rs/types/management_canister_types/src/lib.rs#L2525): it adds the size of the two `u64` (16 bytes in total) on top of the log content size.

Reply:
> yes, otherwise it can have unlimited number of entries with zero content size.

that makes a lot of sense - let's see how to phrase it in the spec

Reply:
Wouldn't that just be `SUM { |l.content| + 8 | l <- Older_logs } > 4KiB`?

Reply:
To keep the spec simple, I've decided to refer to canister log size rather than the size of its content and some overhead.